### PR TITLE
[RAG-40] - Docs - langchain migration

### PR DESCRIPTION
**Pull request by mendonk**
Documentation for LangChain migration.

### CODE REVIEW

File: docs/modules/ROOT/pages/migration.adoc [hunk 1]
```diff
@@ -0,0 +1,288 @@
+= Migration
+
+To migrate an existing LangChain application to RAGStack and use RAGStack's pinned, tested version of LangChain instead, you need to uninstall LangChain and install RAGStack.
```
Comment (kerinin) [inline docs/modules/ROOT/pages/migration.adoc:3]: Could we move this to an aside? I feel like the opening here ought to be something like "Migrating an existing LangChain application to RAGStack is super easy - just change your `requirements.txt` to use `ragstack-ai` and you'll get the benefits of RAGStack, such as <...>". Then we can have an aside that says something like "if you run into problems, you uninstalling your current langchain package could help".

**Review by kerinin — APPROVED** [review]

File: docs/modules/ROOT/pages/migration.adoc (hunk 1 above)
Comment (bjchambers) [reply docs/modules/ROOT/pages/migration.adoc:3]: Or pyproject.toml depending on how you created your Python project.

File: docs/modules/ROOT/pages/migration.adoc [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+
+. `pip uninstall langchain`
+. `pip install ragstack-ai --upgrade-strategy="only-if-needed"`
+. `pip list` to see your current list of packages.
+
+An example migration is shown below.
+
+== RAGStack packages
+RAGStack contains the following packages as of version 0.3.1.
+You can replace any of these packages in your existing application with the `ragstack-ai` version.
```
Comment (cbornet) [inline docs/modules/ROOT/pages/migration.adoc:13]: RAGstack doesn't contain these packages. They are transitive dependencies and their final versions will be resolved on the user environment.
I think we should only mention the dependency versions that we pin (langchain, llama-index, astrapy, cassio, ...)
Currently
```toml
[tool.poetry.dependencies]
python = ">=3.9,<3.12"
astrapy = "~0.6.2"
cassio = "~0.1.3"
unstructured = "^0.10"
llama-index = { version = "0.9.14", extras = ["langchain"] }
langchain = { version = "0.0.343", extras = ["openai"] }
langchain-core = "0.0.8"
```

File: docs/modules/ROOT/pages/migration.adoc (hunk 1 above)
Comment (cbornet) [reply docs/modules/ROOT/pages/migration.adoc:3]: Uninstalling lanchain is not really needed.
Installing RAGStack will replace the langchain version that was there previously if any.

File: docs/modules/ROOT/pages/migration.adoc [hunk 3]
```diff
@@ -0,0 +1,288 @@
...
+. This application requires installation of the following packages:
++
+[source,python]
+----
+pip install langchain datasets openai astrapy tiktoken python-dotenv
+----
++
+. You decide you want to use RAGStack's pinned, tested version of LangChain (`langchain-0.0.349`) instead of the latest version of LangChain (`langchain-0.0.350`).
++
+*You must uninstall LangChain first.*
```
Comment (cbornet) [inline docs/modules/ROOT/pages/migration.adoc:166]: I'm really surprised this is needed.

File: docs/modules/ROOT/pages/migration.adoc [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+pip uninstall langchain
+Successfully uninstalled langchain-0.0.350
+----
++
+. Install the `ragstack-ai` package with the `--upgrade-strategy="only-if-needed"` option.
+This ensures pip will not upgrade any packages that are already installed, unless required by the `ragstack-ai` package.
++
+[source,python]
+----
+pip install --upgrade-strategy="only-if-needed"
```
Comment (cbornet) [inline docs/modules/ROOT/pages/migration.adoc:179]:
```suggestion
pip install --upgrade-strategy="only-if-needed" ragstack-ai
```

**Pull request merged**